### PR TITLE
feat(blade): Revised single radio API & Tests - PR3

### PR DESCRIPTION
**Pull request by anuraghazra**
- Added tests for Single Radio
- Revised single Radio API

Prev: #607 
Next: #629 

-----

PR Stack: 

1. #605
2. #607
3. #615 (Current)
4. #629
5. #631

### CODE REVIEW

File: packages/blade/src/components/Radio/Radio.tsx [hunk 1]
```diff
@@ -45,74 +20,51 @@ type RadioProps = {
...
-  validationState,
-}: RadioProps): React.ReactElement => {
+const Radio = ({ value, children, helpText, isDisabled }: RadioProps): React.ReactElement => {
   const groupProps = useRadioGroupContext();
+  const isInsideGroup = !isEmpty(groupProps);
 
-  const _validationState = validationState ?? groupProps?.validationState;
-  const _hasError = _validationState === 'error';
+  if (!isInsideGroup) {
+    throw new Error('[Blade Radio]: Cannot use <Radio /> outside of <RadioGroup />');
```
Comment (chaitanyadeorukhkar) [inline packages/blade/src/components/Radio/Radio.tsx:41]:
```suggestion
    throw new Error('[Blade: Radio]: Cannot use <Radio /> outside of <RadioGroup />.');
```

File: packages/blade/src/components/Radio/__tests__/RadioGroup.native.test.tsx [hunk 2]
```diff
@@ -0,0 +1,208 @@
...
+
+    const radio = getByA11yState({ checked: true });
+    expect(radio.props.value).toBe('apple');
+
+    const radios = getAllByA11yRole('radio');
+    const apple = radios.find((radio) => radio.props.value === 'apple');
+    const mango = radios.find((radio) => radio.props.value === 'mango');
+    const orange = radios.find((radio) => radio.props.value === 'orange');
+
+    // expect(getByRole('radio', { hidden: true, checked: true })).toHaveAttribute('value', 'apple');
```
Comment (chaitanyadeorukhkar) [inline packages/blade/src/components/Radio/__tests__/RadioGroup.native.test.tsx:145]: Remove this?

File: packages/blade/src/components/Radio/__tests__/RadioGroup.native.test.tsx [hunk 3]
```diff
@@ -0,0 +1,208 @@
...
+
+    const radios = getAllByA11yRole('radio');
+    radios.forEach((radio) => {
+      expect(radio.props.accessibilityInvalid).toBeTruthy();
+    });
+  });
+
+  it('should work in uncontrolled mode', () => {
+    const labelText = 'Select fruits';
+    const fn = jest.fn();
```
Comment (chaitanyadeorukhkar) [inline packages/blade/src/components/Radio/__tests__/RadioGroup.native.test.tsx:123]: nit: 
```suggestion
    const onChange = jest.fn();
```

File: packages/blade/src/components/Radio/__tests__/RadioGroup.native.test.tsx [hunk 4]
```diff
@@ -0,0 +1,208 @@
...
+    const { getByA11yState, getAllByA11yRole } = renderWithTheme(
+      <RadioGroup
+        name="fruits"
+        label={labelText}
+        defaultValue="apple"
+        onChange={({ value }) => fn(value)}
```
Comment (chaitanyadeorukhkar) [inline packages/blade/src/components/Radio/__tests__/RadioGroup.native.test.tsx:129]: nit: directly pass `onChange` mock function

```suggestion
        onChange={onChange}
```

File: packages/blade/src/components/Radio/__tests__/RadioGroup.native.test.tsx [hunk 5]
```diff
@@ -0,0 +1,208 @@
...
+    const mango = radios.find((radio) => radio.props.value === 'mango');
+    const orange = radios.find((radio) => radio.props.value === 'orange');
+
+    // expect(getByRole('radio', { hidden: true, checked: true })).toHaveAttribute('value', 'apple');
+
+    expect(apple?.props.accessibilityState.checked).toBeTruthy();
+    expect(fn).not.toBeCalled();
+    fireEvent.press(mango!);
+    expect(mango?.props.accessibilityState.checked).toBeTruthy();
+    expect(fn).toBeCalledWith('mango');
```
Comment (chaitanyadeorukhkar) [inline packages/blade/src/components/Radio/__tests__/RadioGroup.native.test.tsx:151]: nit:
```suggestion
    expect(onChange).toBeCalledWith(({value: 'mango'}));
```

File: packages/blade/src/components/Radio/__tests__/RadioGroup.web.test.tsx [hunk 6]
```diff
@@ -0,0 +1,189 @@
...
+            onChange={({ value }) => {
+              setValue(value);
+              fn(value);
+            }}
+          >
+            <Radio value="apple">Apple</Radio>
+            <Radio value="mango">Mango</Radio>
+            <Radio value="orange">Orange</Radio>
+          </RadioGroup>
+          <p data-testid="values">{value}</p>
```
Comment (chaitanyadeorukhkar) [inline packages/blade/src/components/Radio/__tests__/RadioGroup.web.test.tsx:168]: What is this for?

**Review by chaitanyadeorukhkar — CHANGES_REQUESTED** [review]

File: packages/blade/src/components/Radio/__tests__/RadioGroup.web.test.tsx (hunk 6 above)
Comment (anuraghazra) [reply packages/blade/src/components/Radio/__tests__/RadioGroup.web.test.tsx:168]: Just a dummy text node to assert to value against the controlled radio.

**Pull request merged**